### ISSUE TITLE
Cannot define property:callSite, object is not extensible

### ISSUE THREAD

**Issue by rleemorlang** [AndreasMadsen/stack-chain#9]
I'm using CucumberJS, Selenium Webdriver, and either of Chromedriver or PhantomJS. If I tell Selenium to do something silly (like set a cookie on a data URL) I get an error like this:

```
      TypeError: Cannot define property:callSite, object is not extensible.
          at Function.defineProperty (native)
          at Object.defineProperty.set (/home/rlm/ws/node_modules/cucumber/node_modules/stack-chain/stack-chain.js:203:14)
          at Function.prepareStackTrace (/home/rlm/ws/node_modules/cucumber/node_modules/stack-chain/stack-chain.js:141:20)
          at promise.Promise.goog.defineClass.resolve_ (/home/rlm/ws/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:1128:38)
          at eval (/home/rlm/ws/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:1059:14)
          at reject (/home/rlm/ws/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:1400:7)
          at eval (/home/rlm/ws/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:2708:36)
          at Array.forEach (native)
          at promise.ControlFlow.goog.defineClass.goog.defineClass.abort_.error.interrupts_.forEach [as abort_] (/home/rlm/ws/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:2708:19)
          at promise.ControlFlow.goog.defineClass.goog.defineClass.abort_.error.executeNext_.processUnhandledRejections_ (/home/rlm/ws/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:2808:12)
```

If I hack `stack-chain.js` to have the following, I can at least see the backtrace for the underlying error. 

```
    if (Object.isExtensible(this)) {
      Object.defineProperty(this, 'callSite', {
          value: frames,
          writable: true,
          configurable: true
      });
    };
```

I know it's not stack-chain's fault that these Error objects aren't extensible, but should stack-chain be checking for this first?

**Comment by AndreasMadsen** [comment]
Sorry, I'm don't know any of those tools. I'm okay with the patch, but you will have to show that it isn't one of those tools which does something strange to the Error object.

**Comment by rleemorlang** [comment]
That's the thing. I'm sure it _is_ some other tool that's doing something strange to the Error. However it seems like stack-chain should defend against this possibility. I'm not familiar enough with everything yet for that to be a strong opinion, merely an idea, which is why I raised it this way.

For a simple example of what I mean that requires only stack-chain, see this gist: https://gist.github.com/rleemorlang/a7f89993e639a7c569be

Would it make sense for stack-chain to check for this and handle it more gracefully? In my real-world case, stack-chain's behaviour is making it extremely difficult to troubleshoot the problem because it's masking the underlying error.

**Comment by AndreasMadsen** [comment]
>  In my real-world case, stack-chain's behaviour is making it extremely difficult to troubleshoot the problem because it's masking the underlying error.

You could also argue that using `isExtensible` hides an underlying error.

I think the best solution is to not use `isExtensible` in the setter, but instead here: https://github.com/AndreasMadsen/stack-chain/blob/master/stack-chain.js#L122

That way if anyone else sets `callSite`, they will get an `object is not extensible` error which is what should happen. But `stack-chain` will also be gracefully as you suggests.

Do you wan't to make a pull request?

**Comment by rleemorlang** [comment]
That makes sense to me. I'll put together a PR.

In the meantime I've isolated the problem. Summarizing for anyone who might find it useful:

[CucumberJs](https://github.com/cucumber/cucumber-js) requires stack-chain. An optional but common use of Cucumber involves requiring [Selenium Webdriver](https://github.com/SeleniumHQ/selenium). Selenium requires Google's [Closure Library](https://github.com/google/closure-library). Selenium uses Closure's `goog.defineClass()` method to define some errors, and Closure, by default, calls `Object.seal()` on these instances.

**Comment by AndreasMadsen** [comment]
Fixed by #10
